python/ray/experimental/gpu_object_manager/gpu_object_manager.py

```diff
@@ -296,34 +354,31 @@ def add_gpu_object_ref(
             tensor_transport_meta: The tensor transport metadata that is pre-computed.
                 This is known at ref creation time if the object is created through ray.put.
         """
-        from ray.experimental.gpu_object_manager.gpu_object_store import (
-            __ray_get_tensor_transport_metadata__,
+        self.set_gpu_object_metadata(
+            obj_ref.hex(),
+            GPUObjectMeta(
+                src_actor=src_actor,
+                tensor_transport_backend=tensor_transport,
+                tensor_transport_meta=tensor_transport_meta,  # None if not from ray.put
+                sent_dest_actors=set(),
+                sent_to_src_actor_and_others_warned=False,
+            ),
         )
 
-        obj_id = obj_ref.hex()
-        if not tensor_transport_meta:
-            # Submit a Ray actor task to the source actor to get the tensor metadata.
-            # The metadata is a list of tuples, where each tuple contains the shape and dtype
-            # of a tensor in the GPU object store. This function returns an ObjectRef that
-            # points to the tensor metadata.
-            # NOTE(swang): We put this task on the background thread to avoid tasks
-            # executing on the main thread blocking this task.
-            tensor_meta = src_actor.__ray_call__.options(
-                concurrency_group="_ray_system"
-            ).remote(__ray_get_tensor_transport_metadata__, obj_id, tensor_transport)
-        else:
-            tensor_meta = tensor_transport_meta
-        self.managed_gpu_object_metadata[obj_id] = GPUObjectMeta(
-            src_actor=src_actor,
-            tensor_transport_backend=tensor_transport,
-            tensor_transport_meta=tensor_meta,
-            sent_dest_actors=set(),
-            sent_to_src_actor_and_others_warned=False,
+    def set_tensor_transport_metadata_and_trigger_queued_transfers(
+        self, object_id: str, tensor_transport_meta: "TensorTransportMetadata"
+    ):
+        """
+        Sets the tensor transport metadata for an object and triggers any queued
+        up transfers for that object.
+        """
+        dst_actors = self.set_tensor_transport_metadata_and_pop_queued_transfers(
```

Comment:
Should we just squash this function into the current one? Seems it's not used elsewhere.